client/js/components/procedure/admin/AdministrationCustomFieldsList.vue

```diff
@@ -0,0 +1,401 @@
+    checkIfNameIsUnique (name) {
+      const identicalNames = Object.values(this.customFields).filter(field => field.attributes.name === name)
+
+      return identicalNames.length <= 1
+    },
+
+    /**
+     * @param options { array } Array of strings
+     * @param name { string }
+     * @returns { boolean }
+     */
+    checkIfOptionNameIsUnique (options, name) {
+      const identicalNames = options.filter(optionName => optionName === name)
+
+      return identicalNames.length <= 1
+    },
+
+    /**
+     * Fetch custom fields that are available either in the procedure or in the procedure template
+     */
+    fetchCustomFields () {
+      this.isInitiallyLoading = true
+
+      const sourceEntity = this.isProcedureTemplate
+        ? 'ProcedureTemplate'
+        : 'AdminProcedure'
+
+      const payload = {
+        id: this.procedureId,
+        fields: {
+          [sourceEntity]: [
+            'segmentCustomFields'
+          ].join(),
+          CustomField: [
+            'name',
+            'description',
+            'options'
+          ].join()
+        },
+        include: ['segmentCustomFields'].join()
+      }
+
+      this.getCustomFields(payload)
+        .catch(err => console.error(err))
+        .finally(() => {
+          this.isInitiallyLoading = false
+        })
+    },
+
+    hideOptions (rowData) {
+      const field = this.customFieldsReduced.find(field => field.id === rowData.id)
+
+      if (field) {
+        field.open = false
+      }
+    },
+
+    removeOptionInput (index) {
+      this.newFieldOptions.splice(index, 1)
+    },
+
+    resetNewFieldForm () {
+      this.newFieldOptions = [
+        '',
+        ''
+      ]
+    },
+
+    /**
+     * Prepare payload and send create request for custom field
+     * @param customFieldData {Object}
+     * @param customFieldData.name {String}
+     * @param customFieldData.description {String}
+     */
+    saveNewField (customFieldData) {
+      const { description, name } = customFieldData
+      const options = this.newFieldOptions.filter(option => option !== '')
+      const isDataValid = this.validateNamesAreUnique(name, options)
+
+      if (!isDataValid) {
+        return
+      }
+
+      this.isLoading = true
+
+      const payload = {
+        type: 'CustomField',
+        attributes: {
+          description,
+          name,
+          options,
+          sourceEntity: this.isProcedureTemplate ? 'PROCEDURE_TEMPLATE' : 'PROCEDURE',
+          sourceEntityId: this.procedureId,
+          targetEntity: 'SEGMENT',
+          fieldType: 'singleSelect'
+        }
+      }
+
+      this.createCustomField(payload)
+        .then(() => {
+          this.isSuccess = true
+          dplan.notify.confirm(Translator.trans('confirm.saved'))
+        })
+        .catch(err => {
+          console.error(err)
+        })
+        .finally(() => {
+          this.isLoading = false
+          this.resetNewFieldForm()
+          this.fetchCustomFields()
+        })
```

Comment:
why does this lives here in the list and not in the create form? 
If this would be in the create form, we could remove the watcher and the code is where the interaction happens

Reply:
I think the concern is very valid and we should move it. Originally the idea was to use the create form also in the places view, since the interface is very similar. In the end I decided against it though because we don't consider places to be custom fields, and the interfaces are probably going to develop into different directions over time. So the form component also doesn't need to have a slot anymore and the option inputs can be moved into it, too, but I'd prefer to refactor that after merging this PR, as it might delay the feature freeze otherwise.